### PR TITLE
feat: schema-safe access for non-pacs.002 support

### PR DESCRIPTION
**Pull request by ReebaPaysys**
# SPDX-License-Identifier: Apache-2.0

## What did we change?

1.  Schema-safe access + validation

-  Added a new utility, createSafeObjectFromEndpoint, that provides schema-driven, fail-fast, dot-notation access into transaction payloads (with type coercion and a Redis bootstrap fallback).
- Added comprehensive tests for this behavior:
   - __tests__/safeObjectFromSchema.test.ts
   - __tests__/safeObjectFromSchema.redisBootstrap.test.ts
- Updated README.md with detailed documentation, runtime guarantees, usage examples, and compatibility notes for schema-safe access.

2. Database manager testing

- Added extensive tests around database manager methods, specifically saveAccount and updateCondition, validating:
   -  SQL structure
   - Parameter ordering
   - Type handling
   - Edge cases like timestamp formats and partial field updates.

3. Type guard validation

-  Added tests for transaction type guards:
     - isBaseMessageTransaction
     - isPacs002Transaction
- Ensures correct identification of valid transaction shapes and correct rejection of invalid/malformed ones.

4. Documentation + configuration updates
- Updated README.md and VARIABLES.md for clarity and correctness, including fixing variable typing such as ENRICHMENT_DATABASE_PORT from String to Number.
- Updated .npmrc to use a template variable for GH_TOKEN for better environment compatibility.

5. Test suite maintenance
- Added utility imports in utils.test.ts to improve coverage.

## Why are we doing this?

- To make payload access safer and more predictable at runtime by enforcing schemas and failing fast instead of allowing silent undefined/wrong-type values when traversing nested payloads.
- To improve confidence and prevent regressions in database operations by verifying SQL shape, parameter order, and tricky edge cases (timestamps, selective updates).
- To ensure transaction handling logic is more robust by validating type guards with explicit tests, reducing the risk of misclassification and downstream runtime errors.
- To reduce configuration/documentation drift by aligning docs and env var types with how the system actually expects them to behave, and making auth/token configuration more portable across environments.

## How was it tested?
- [ ] Locally
- [ ] Development Environment
- [ ] Not needed, changes very basic
- [ ] Husky successfully run
- [ ] Unit tests passing and Documentation done

## Summary by CodeRabbit

* **New Features**
  * Schema-enforced dot-notation access to transaction payloads; support for a new BaseMessage transaction shape and safer payload (de)serialization.

* **Documentation**
  * Added schema-safe access usage, endpoint path contract, fail-fast guarantees, and a temporary Redis bootstrap note.

* **Updates**
  * Corrected enrichment DB environment variable docs (port type and cert optionality).

* **Tests**
  * Added/expanded tests for schema bootstrapping, safe-access behavior, type guards, and message buffer serialization.

* **Chores**
  * Package version and dev publish script updated.

### CODE REVIEW

File: src/helpers/proto/Full.proto [hunk 1]
```diff
@@ -349,6 +353,10 @@ message FRMSMessage {
         Fitoficstmrcdttrf FIToFICstmrCdtTrf = 3;
         Cstmrcdttrfinitn CstmrCdtTrfInitn = 4;
         Cdtrpmtactvtnreq CdtrPmtActvtnReq = 5;
+        string MsgId = 7;
+        Payload1 Payload = 8;
+        string endpointPath = 9;
+        dataCache DataCache = 10;
```
[the remark this answers is not in the document]
Comment (ReebaPaysys) [reply src/helpers/proto/Full.proto:359]: This is for testing. I have removed this.

File: src/helpers/transactionTypeGuards.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
+import type { BaseMessage, Pacs002 } from '../interfaces';
+
+const isRecord = (value: unknown): value is Record<string, unknown> => typeof value === 'object' && value !== null;
+
```
[the remark this answers is not in the document]
Comment (ReebaPaysys) [reply src/helpers/transactionTypeGuards.ts:6]: Not applicable. The behavior of isRecord is tested and works as intended.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@tazama-lf/frms-coe-lib",
-  "version": "7.0.2-rc.2",
+  "version": "0.0.1-psl.3",
```
[the remark this answers is not in the document]
Comment (ReebaPaysys) [reply package.json]: This is intended for build 1.1, frms-coe-lib version 0.0.1-psl is intended to start moving tazama ecosystem to non-pacs002.
